### PR TITLE
Handle Django GENERATE_ENUMS_FROM_CHOICES with strawberry.auto

### PR DESCRIPTION
**Pull request by pcraciunoiu**
## Description

It seems that `GENERATE_ENUMS_FROM_CHOICES` breaks `strawberry.auto` types with this error:

```
TypeError: DjangoModelStrawberryType fields cannot be resolved. 'NoneType' object is not iterable
```

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: strawberry_django/fields/types.py [hunk 1]
```diff
@@ -435,7 +435,7 @@ def resolve_model_field_type(
     elif (
         settings["GENERATE_ENUMS_FROM_CHOICES"]
         and isinstance(model_field, Field)
-        and hasattr(model_field, "choices")
+        and getattr(model_field, "choices", None)
```
Comment (pcraciunoiu) [inline strawberry_django/fields/types.py:438]: In my testing I found that `model_field.choices` was set to `None` even for `CharField`

**Comment by bellini666** [comment]
@pcraciunoiu nice! :)

I just noticed that we don't have a test for this, and that's probably the reason this slipped during the merge.

Could you add a test for it just to make sure it doesn't break again in the future?

**Review by bellini666 — CHANGES_REQUESTED** [review]
Looks great!

Just need to fix the linting stuff that broke from `pre-commit`

**Comment by pcraciunoiu** [comment]
> Looks great!
> 
> Just need to fix the linting stuff that broke from `pre-commit`

Sync! Pushed just now

**Review by bellini666 — APPROVED** [review]

**Pull request merged**